plugins/modules/inventory_intent.py

```diff
@@ -1335,43 +1348,85 @@ def resync_devices(self):
 
         device_ids = self.get_device_ids(input_device_ips)
         try:
+            # Resync the device in a batch of 200 devices at a time in inventory by default
+            start = 0
+            resync_failed_for_all_device = False
+            resync_device_count = self.config[0].get("resync_device_count", 200)
+            device_resync_list, resync_failed_devices = [], []
             force_sync = self.config[0].get("force_sync", False)
-            resync_param_dict = {
-                'payload': device_ids,
-                'force_sync': force_sync
-            }
-            response = self.dnac._exec(
-                family="devices",
-                function='sync_devices_using_forcesync',
-                op_modifies=True,
-                params=resync_param_dict,
-            )
-            self.log("Received API response from 'sync_devices_using_forcesync': {0}".format(str(response)), "DEBUG")
+            resync_task_dict = {}
+
+            while start < len(device_ids):
+                device_ids_list = device_ids[start:start + resync_device_count]
+                device_ips_list = input_device_ips[start:start + resync_device_count]
+                resync_param_dict = {
+                    'payload': device_ids_list,
+                    'force_sync': force_sync
+                }
+                self.log("Request payload for reysnc Device having the device ids: {0}".format(device_ids_list), "INFO")
+                response = self.dnac._exec(
+                    family="devices",
+                    function='sync_devices_using_forcesync',
+                    op_modifies=True,
+                    params=resync_param_dict,
+                )
+                self.log("Received API response from 'sync_devices_using_forcesync': {0}".format(str(response)), "DEBUG")
+
+                if not response or not isinstance(response, dict):
+                    self.status = "failed"
+                    self.msg = "Unable to resync the device(s) {0} in the inventory as response is empty.".format(device_ips_list)
+                    self.log(self.msg, "ERROR")
+                    self.result['response'] = self.msg
+                    return self
 
-            if response and isinstance(response, dict):
                 task_id = response.get('response').get('taskId')
+                resync_task_dict[task_id] = device_ips_list
+                start += resync_device_count
+
+            for task_id, device_list in resync_task_dict.items():
+                max_timeout = self.config[0].get("resync_max_timeout", 600)
+                start_time = time.time()
+
+                while (True):
+                    end_time = time.time()
```

Comment:
this end_time local variable not required. We can directly write as :

    if time.time() - start_time >= max_timeout: